flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/codegen/calls/BridgingFunctionGenUtil.scala

```diff
@@ -147,7 +147,8 @@ object BridgingFunctionGenUtil {
         externalOperands,
         returnType)
     } else {
-      generateScalarFunctionCall(ctx, functionTerm, externalOperands, outputDataType)
+      generateScalarFunctionCall(ctx, functionTerm, externalOperands,
```

Comment:
I think the right way to remove `copy` is in above `prepareExternalOperands`.
- We should check `ExternalGeneratedExpression` in `genToExternalConverterAll`.
- Then remove `case external: ExternalGeneratedExpression` in `generateScalarFunctionCall`, we have already generated the `externalOperands`!

Reply:
I quite agree with you, I modify the code and please help to review it